include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -666,6 +747,35 @@ __pattern_remove_if(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec,
     return __copy_last_id;
 }
 
+#if _ONEDPL_CPP20_RANGES_PRESENT
+template <typename _BackendTag, typename _ExecutionPolicy, typename _R, typename _Proj, typename _Pred>
+std::ranges::borrowed_subrange_t<_R>
+__pattern_remove_if(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _R&& __r, _Pred __pred, _Proj __proj)
+{
+    auto __pred_1 = [__pred, __proj](auto&& __val)
+        { return std::invoke(__pred, std::invoke(__proj, std::forward<decltype(__val)>(__val)));};
+
+    auto __n = std::ranges::size(__r);
+    auto __idx = oneapi::dpl::__internal::__ranges::__pattern_remove_if(__tag, std::forward<_ExecutionPolicy>(__exec),
```

Comment:
As we discussed offline, pattern `__pattern_remove_if` uses `__brick_copy`, but `remove_if` and `remove` do not require copyability. It should be either fixed, or reflected as a known limitation.